pkgs/by-name/co/compiler-explorer-cpp/package.nix

```diff
@@ -0,0 +1,204 @@
+{ lib
+, stdenv
+, runCommand
+, makeWrapper
+, python3Full
+, rsync
+, writeText
+, compiler-explorer-base
+, gcc48, gcc10, gcc13
+, clang_9, clang_14, clang_17 
+, lld_17
+, llvmPackages_17
+}:
+let
+
+  defaultGcc = gcc13;
+  defaultClang = clang_17;
+
+  gccEntries = [gcc48 gcc10 gcc13];
+  clangEntries = [clang_9 clang_14 clang_17];
+
+  attrToDot =
+    let
+      attrToDotHelper = prefix: x:
+        lib.concatStrings
+          (lib.mapAttrsToList
+            (k: v:
+              if builtins.elem (builtins.typeOf v) [ "string" "int" "float" ]
+              then "${prefix}${k}=${toString v}\n"
+              else attrToDotHelper "${prefix}${k}." v)
+            x);
+    in attrToDotHelper "";
```

Comment:
It would be nice to avoid duplicating this. One way is by moving all of the etc files into one package. Alternatively, the passthru argument to mkDerivation can be used to export the helper function.